### ISSUE TITLE
Require SSL false and certificate error while setting up sample - Exception accessing: /identity/connect/consent System.ArgumentNullException: Value cannot be null. Parameter name: certificate

### ISSUE THREAD

**Issue by raxitr** [IdentityServer/IdentityServer3#1516]
I am trying out embedded IdSvr sample (http://identityserver.github.io/Documentation/docs/overview/mvcGettingStarted.html) and running into following issues,

I have set Requires SSL to false to avoid setting SSL since I am trying out samples everthing works fine but after logging I get "There was an unexpected error." instead of displaying about view.

below is the only change I made to startup to turn off SSL

idsrvApp.UseIdentityServer(new IdentityServerOptions
                {
                    SiteName = "Embedded IdentityServer",
                    
                    RequireSsl = false,
                    //SigningCertificate = LoadCertificate(),

                    Factory = InMemoryFactory.Create(
                        users: Users.Get(),
                        clients: Clients.Get(),
                        scopes: StandardScopes.All)
                });

 Tracing has following error but not sure how to fix since I do not use certificate

iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.Endpoints.AuthorizeEndpointController]: 6/20/2015 7:31:50 PM +00:00 -- Showing consent screen
iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.Endpoints.AuthorizeEndpointController]: 6/20/2015 7:31:50 PM +00:00 -- End authorize request
iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.Endpoints.AuthorizeEndpointController]: 6/20/2015 7:31:56 PM +00:00 -- Resuming from consent, restarting validation
iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.Validation.AuthorizeRequestValidator]: 6/20/2015 7:31:56 PM +00:00 -- Start authorize request protocol validation
iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.Validation.AuthorizeRequestValidator]: 6/20/2015 7:31:56 PM +00:00 -- Authorize request validation success
 {
  "ClientId": "mvc",
  "ClientName": "MVC Client",
  "RedirectUri": "http://localhost:60827/",
  "AllowedRedirectUris": [
    "http://localhost:60827/"
  ],
  "SubjectId": "1",
  "ResponseType": "id_token",
  "ResponseMode": "form_post",
  "Flow": "Implicit",
  "RequestedScopes": "openid profile",
  "State": "OpenIdConnect.AuthenticationProperties=BXnN5fm1gOSCNqAT7mzFri1u6NShrXsUDI9aN8-peIXA1mwLennfkxZ1W4_fVGW3nSVDIlod4sZgMU6eIZG_kiYoQUk-ZLLrz6v8fNERF-fNwBFKcsEfhHX8aVy4kUBwBR7RHOXSET8IJzsA9lPsN2M-ivkdRBDot6lZtm_44nGNwxd2ZPROMp6s8xSRkchx5x2yUSiTNdxP8l1x0qNcfw",
  "Nonce": "635704254957862922.Zjg3YTQwNzktNjllZS00NzI5LTkzNTctMGIxZTY1ZTI0YmFkZTk1OTBmZGMtMWI2Yy00YzNiLWE3NTUtOTczYTQ2YmQ1ZjQx",
  "SessionId": "9a302a9f984782adae067b2f97b19cc1",
  "Raw": {
    "client_id": "mvc",
    "redirect_uri": "http://localhost:60827/",
    "response_mode": "form_post",
    "response_type": "id_token",
    "scope": "openid profile",
    "state": "OpenIdConnect.AuthenticationProperties=BXnN5fm1gOSCNqAT7mzFri1u6NShrXsUDI9aN8-peIXA1mwLennfkxZ1W4_fVGW3nSVDIlod4sZgMU6eIZG_kiYoQUk-ZLLrz6v8fNERF-fNwBFKcsEfhHX8aVy4kUBwBR7RHOXSET8IJzsA9lPsN2M-ivkdRBDot6lZtm_44nGNwxd2ZPROMp6s8xSRkchx5x2yUSiTNdxP8l1x0qNcfw",
    "nonce": "635704254957862922.Zjg3YTQwNzktNjllZS00NzI5LTkzNTctMGIxZTY1ZTI0YmFkZTk1OTBmZGMtMWI2Yy00YzNiLWE3NTUtOTczYTQ2YmQ1ZjQx"
  }
}
iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.ResponseHandling.AuthorizeResponseGenerator]: 6/20/2015 7:31:56 PM +00:00 -- Creating Implicit Flow response.
Debug: [Thinktecture.IdentityServer.Core.Services.Default.DefaultTokenService]: 6/20/2015 7:31:56 PM +00:00 -- Creating identity token
iisexpress.exe Information: 0 : [Thinktecture.IdentityServer.Core.Services.Default.DefaultClaimsProvider]: 6/20/2015 7:31:56 PM +00:00 -- Getting claims for identity token for subject: 1
Debug: [Thinktecture.IdentityServer.Core.Services.Default.DefaultTokenService]: 6/20/2015 7:31:56 PM +00:00 -- Creating JWT identity token
iisexpress.exe Error: 0 : [Thinktecture.IdentityServer.Core.Configuration.Hosting.ErrorPageFilterAttribute]: 6/20/2015 7:32:00 PM +00:00 -- Exception accessing: /identity/connect/consent
System.ArgumentNullException: Value cannot be null.
Parameter name: certificate
   at System.IdentityModel.Tokens.X509SecurityToken..ctor(X509Certificate2 certificate, String id, Boolean clone, Boolean disposable)
   at System.IdentityModel.Tokens.X509SigningCredentials..ctor(X509Certificate2 certificate)
   at Thinktecture.IdentityServer.Core.Services.Default.DefaultTokenSigningService.SignTokenAsync(Token token) in c:\ballen\github\identity\IdSrv3\IdentityServer3\source\Core\Services\Default\DefaultTokenSigningService.cs:line 55
   at Thinktecture.IdentityServer.Core.Services.Default.DefaultTokenService.<CreateSecurityTokenAsync>d__c.MoveNext() in c:\ballen\github\identity\IdSrv3\IdentityServer3\source\Core\Services\Default\DefaultTokenService.cs:line 210
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Thinktecture.IdentityServer.Core.ResponseHandling.AuthorizeResponseGenerator.<CreateImplicitFlowResponseAsync>d__18.MoveNext() in c:\ballen\github\identity\IdSrv3\IdentityServer3\source\Core\ResponseHandling\AuthorizeResponseGenerator.cs:line 169
--- End of stack trace from previous location where exception was thrown ---

**Comment by leastprivilege** [comment]
You need a cert to sign tokens. That's unrelated to ssl.

**Comment by raxitr** [comment]
Thank You, I will get the certificate installed and add the SigningCertificate = LoadCertificate() call again.

**Comment by raxitr** [comment]
All sorted and working fine. Thanks for your help. 

Couple of points I wish to mention to help out others like me who are starting out with IdServer3, the instruction given are fairly simple and easy to follow until you have run into certificate issues, following is what I ran into

1) I have installed certificates (pfx files) as instructed, placed in Personal/ Trusted Root CA  but ended up getting "A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider."

2) Observing thumb print closely I figured its the IIS Express certificate having trust issues. I have then installed the IIS Express certificate into Trusted Root CA of local computer and everything worked as expected.
